src/ProjectTemplates/test/BaselineTest.cs

```diff
@@ -116,32 +103,61 @@ public async Task Template_Produces_The_Right_Set_Of_FilesAsync(string arguments
             }
         }
 
-        private string SanitizeArgs(string arguments)
+        private static ConcurrentDictionary<string, object> _projectKeys = new();
+
+        private string CreateProjectKey(string arguments)
         {
-            var text = TemplateNameRegex.Match(arguments)
-                .Groups.TryGetValue("template", out var template) ? template.Value : "";
+            var text = "baseline";
 
-            text += AuthenticationOptionRegex.Match(arguments)
-                .Groups.TryGetValue("auth", out var auth) ? auth.Value : "";
+            // Turn string like "new templatename -minimal -au SingleOrg --another-option OptionValue"
+            // into array like [ "new templatename", "minimal", "au SingleOrg", "another-option OptionValue" ]
+            var argumentsArray = arguments
+                .Split(" --", int.MaxValue, StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
+                .SelectMany(e => e.Split(" -", StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries))
+                .ToArray();
```

Comment:
Is this any different than the following?

```suggestion
            var argumentsArray = arguments
                .Split(new[] { " --", " -" }, StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
                .ToArray();
```

Reply:
So I tried that first and it didn't seem to produce what I wanted, but it may be that my intent changed during refactoring. I can try again although I found the meaning of passing a string array as slightly ambiguous WRT to pass ordering.

Reply:
My basic testing shows it gives the same result "new templatename -minimal -au SingleOrg --another-option OptionValue". I nerd sniped @BrennanConroy who found this in the docs.

> To avoid ambiguous results when strings in separator have characters in common, the Split method proceeds from the beginning to the end of the value of the instance, and matches the first element in separator that is equal to a delimiter in the instance. The order in which substrings are encountered in the instance takes precedence over the order of elements in separator.

https://docs.microsoft.com/en-us/dotnet/api/system.string.split?view=net-5.0#System_String_Split_System_String___System_StringSplitOptions_

So it seems it should work.

Reply:
Changed it locally and it does work. I'll roll it into the change to make `WeatherForecast` a record too.